Library/Homebrew/utils/github.rb

```diff
@@ -309,8 +309,8 @@ def repository(user, repo)
     open_api(url_to("repos", user, repo))
   end
 
-  def search_code(**qualifiers)
-    matches = search("code", **qualifiers)
+  def search_code(user: "Homebrew", path: ["Formula", "Casks", "."], filename: nil, extension: "rb")
```

Comment:
I think these defaults except `filename: nil` are better kept in the `search.rb`

Reply:
I don't think it makes sense to use keyword arguments here since this should support all fields supported by GitHub search. Also I think there may definitely be some places in other repos where `repo` is used.

Reply:
I think `Homebrew::Search.search_taps` can pass only `query` to `GitHub.search_code` and it easier to read when I set default args.

Reply:
While I couldn't seem to find usage of `search_code` with the `repo` field, I agree with @reitermarkus that the method should support all fields that GitHub's search does. We probably shouldn't be removing the ability to use these keys based on just the usage in `search_taps`.